bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/configuration/CustomBodyMenuConfiguration.java

```diff
@@ -0,0 +1,137 @@
+import org.eclipse.swt.widgets.MenuItem;
+
+/**
+ * Configures the context menu for the body region of the NatTable.
+ * 
+ * @author Stephan Druskat (mail@sdruskat.net)
+ */
+public class CustomBodyMenuConfiguration extends AbstractUiBindingConfiguration {
+
+  private static final String DELETE_CELL_ITEM = "DELETE_CELL_ITEM"; //$NON-NLS-1$
+  private Menu menu;
+  private final NatTable table;
+  private final SelectionLayer selectionLayer;
+
+  /**
+   * Constructor setting the table and selection layer fields, and creating the menu via
+   * {@link #createMenu()}.
+   * 
+   * @param table The NatTable upon whose body region the menu should be constructed
+   * @param selectionLayer The SelectionLayer to be used for configuring menu item states based on
+   *        selection
+   */
+  public CustomBodyMenuConfiguration(NatTable table, SelectionLayer selectionLayer) {
+    this.table = table;
+    this.selectionLayer = selectionLayer;
+    this.menu = createMenu();
+  }
+
+  private Menu createMenu() {
+    PopupMenuBuilder builder = new PopupMenuBuilder(this.table);
+    builder.withMenuItemProvider(DELETE_CELL_ITEM, new DeleteItemProvider());
+    builder.withVisibleState(DELETE_CELL_ITEM, new ValidSelectionState());
+    Menu popUpMenu = builder.build();
+    return popUpMenu;
+  }
+
+  @Override
+  public void configureUiBindings(UiBindingRegistry uiBindingRegistry) {
+    uiBindingRegistry.registerMouseDownBinding(
+        new MouseEventMatcher(SWT.NONE, GridRegion.BODY, MouseEventMatcher.RIGHT_BUTTON),
+        new PopupMenuAction(this.menu));
+
+  }
+
+  /**
+   * Provides a menu item for deleting cells.
+   * 
+   * @author Stephan Druskat (mail@sdruskat.net)
+   */
+  public class DeleteItemProvider implements IMenuItemProvider {
+
+
+    @Override
+    public void addMenuItem(NatTable natTable, Menu popupMenu) {
+      MenuItem item = new MenuItem(popupMenu, SWT.PUSH);
+      item.setText("Delete cell(s)");
+      item.setEnabled(true);
+      item.addSelectionListener(new SelectionAdapter() {
+        @Override
+        public void widgetSelected(SelectionEvent event) {
+          new DeleteSelectionAction().run(natTable, null);
+        }
+      });
+    }
+
+  }
+
+  /**
+   * A menu item state based on valid selection of cells.
+   * 
+   * <p>
+   * {@link #isActive(NatEventData)} returns <code>true</code> only when cells are selected, and no
+   * cells containing token text are selected.
+   * </p>
+   * 
+   * @author Stephan Druskat (mail@sdruskat.net)
+   */
+  private class ValidSelectionState implements IMenuItemState {
+
+    @Override
+    public boolean isActive(NatEventData natEventData) {
+      if (selectionLayer.getSelectedCells().isEmpty()) {
+        return false;
+      } else {
+        PositionCoordinate[] selectedCellCoordinates = selectionLayer.getSelectedCellPositions();
+
+        for (PositionCoordinate coord : selectedCellCoordinates) {
+          if (coord.getColumnPosition() == 0) {
```

Comment:
While it is implicitly clear because of the Javadoc on the function, it might help to comment why this is zero (we don't have a variable or other clearly named check for the token column and you need to know about the column index semantics).
```suggestion
          // The first column is reserved for tokens
          if (coord.getColumnPosition() == 0) {
```